python/hidet/distributed/store.py

```diff
@@ -0,0 +1,210 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List, Optional
+from datetime import timedelta, datetime
+import time
+import struct
+import os
+import atexit
+import filelock
+
+
+class Store:
+    def set(self, key: str, value: bytes) -> None:
+        raise NotImplementedError()
+
+    def get(self, key: str) -> bytes:
+        raise NotImplementedError()
+
+    def add(self, key: str, amount: int) -> int:
+        raise NotImplementedError()
+
+    def compare_set(self, key: str, expected: bytes, desired: bytes) -> bytes:
+        raise NotImplementedError()
+
+    def wait(self, keys: List[str], timeout: Optional[timedelta] = None) -> None:
+        raise NotImplementedError()
+
+    def num_keys(self) -> int:
+        raise NotImplementedError()
+
+    def delete_key(self, key: str) -> bool:
+        raise NotImplementedError()
+
+    def set_timeout(self, timeout: timedelta):
+        raise NotImplementedError()
+
+
+class FileStore(Store):
+    REGULAR_PREFIX = '+'
+    DELETE_PREFIX = '-'
```

Comment:
Hi @soodoshll, do you mind documenting several lines to describe the format of the file used to store the key-value store?